### ISSUE TITLE
RichTextArea from Vaadin 8 is not editable with Firefox ESR 45 and Firefox 51

### ISSUE THREAD

**Issue by qwasli** [vaadin/framework#8851]
The RichTextArea is not editable when opening with Firefox. Tested with Vaadin 8.0.2
Tested Firefox Version: 

- Firefox ESR 45.8.0 (64 bit linux)
- Firefox ESR 45.7.0 (windows)
- Firefox 51.0.1 (64-bit linux)

Firefox 52 (windows) seems to work.

Can be tested here: 
http://demo.vaadin.com/sampler/#ui/data-input/text-input/rich-text-area

**Comment by qwasli** [comment]
I digged somewhat deeper and am pretty sure that by default the widgetset is only compiled for safari if not explicitly set in the widgetset xml like this: 
`<set-property name="user.agent" value="gecko1_8,safari" />`
So it is possible that there are even more problem with Firefox.
Since the precompiled widgetset from vaadin and the autogenerated widgetset xml also has the problem the only workaroud i see is to create a custom widgetset xml for the application and insert the line I mentioned above.

**Comment by Artur-** [comment]
`<set-property name="user.agent" value="gecko1_8,safari" />` is already in `Vaadin.gwt.xml` so setting it again should have no effect. Maybe you are including some add-on which changes the property?

**Comment by qwasli** [comment]
Yes i know that it is in `Vaadin.gwt.xml` but somehow it does not have the wanted effect.  I tested it with a newly created project without any add-ons.  It only works if I add widgetset xml like this: 
```
<module>
    <inherits name="com.vaadin.DefaultWidgetSet"/>
    <set-property name="user.agent" value="gecko1_8,safari" />
</module>
```
I haven't found out yet where the user agent from Vaadin.gwt.xml is overwritten. Could it have something to do with this in Vaadin.gwt.xml ?

```
    <!-- If no proper user agent is found, at least try some, e.g. with crawlers -->
    <set-property-fallback name="user.agent" value="safari" />
```

**Comment by qwasli** [comment]
I think I found the reason for this behavior.
[DefaultWidgetset.gwt.xml](https://github.com/vaadin/framework/blob/master/client/src/main/resources/com/vaadin/DefaultWidgetSet.gwt.xml) inherits Vaadin and Elemental.
```
	<inherits name="com.vaadin.Vaadin" />
	<inherits name="elemental.Elemental" />
```
[Vaadin.gwt.xml](https://github.com/vaadin/framework/blob/master/client/src/main/resources/com/vaadin/Vaadin.gwt.xml) sets `user.agent` as "gecko1_8,safari" but [Elemtal.gwt.xml](https://github.com/vaadin/gwt/blob/master/elemental/src/elemental/Elemental.gwt.xml) sets it to "safari" only.

**Comment by fhaman** [comment]
We experience the same behaviour with Vaadin 8.1.1 using Firefox 55.0.2 (64-bit, Linux) and 52.3.0 (32-Bit, Linux).
The proposed solution by qwasli in comment #2 does not work for us. It works in Chrome 60.0.3112.101 (Official Build) (64-bit, Linux) and in IE 11 under Windows 8.
It would be nice if this could be fixed as soon as possible.
Thank you!

**Comment by wbadam** [comment]
This issue is addressed in https://github.com/vaadin/framework/issues/10469 and the patch is released with 8.2.1
